python/pyspark/sql/dataframe.py

```diff
@@ -300,7 +300,7 @@ def take(self, num):
         >>> df.take(2)
         [Row(age=2, name=u'Alice'), Row(age=5, name=u'Bob')]
         """
-        return self.limit(num).collect()
+        return self.coalesce(1).limit(num).collect()
```

Comment:
Is it possible to just call scala DataFrame.take and get the result? You lose the socket thing if the number of rows is enormous, but that doesn't seem like a big deal for take.

Note that this changes the behavior of take in Python vs Scala. In Scala, you can still get parallelism, whereas in Python you coalesced it into a single partition, and as a result the degree of parallelism is now just 1.